frontend/src/register.tsx

```diff
@@ -0,0 +1,135 @@
+import {Bounce, toast, ToastContainer} from "react-toastify";
+import 'react-toastify/dist/ReactToastify.css';
+import React from "react";
+import {useNavigate} from "react-router";
+
+
+// Define the interface for the request payload
+interface UserPayload {
+    username: string;
+    email: string;
+    firstName: string;
+    lastName: string;
+}
+
+function Register() {
+
+  const navigate = useNavigate();
+// Function to create a new user
+    async function createUser(payload: UserPayload): Promise<void> {
+        const url = 'http://localhost:5000/users/new';
+
+        try {
+            const response = await fetch(url, {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+                body: JSON.stringify(payload),
+            });
+
+            if (!response.ok) {
+                // Handle response errors
+                const errorResponse = await response.json()
+                toast.error(errorResponse.error);
+                console.error(response)
+                return
+                // throw new Error(`Error: ${response.status} - ${errorRes}`);
+            }
+
+            const data = await response.json();
+            console.log('User created successfully:', data);
+            toast.success("Success! Redirecting home !");
+
+            setTimeout(() => {
+                navigate("/");
+            }, 3000)
+
+        } catch (error) {
+            console.error('Error creating user:', error);
+            toast.error(JSON.stringify(error));
+            console.log(error)
+
+        }
+    }
+
+    const handleSubmit = async (e: React.SyntheticEvent) => {
+        e.preventDefault()
+
+        const usernameInput = document.getElementById('username') as HTMLInputElement | null;
+        const emailInput = document.getElementById('email') as HTMLInputElement | null;
+        const firstNameInput = document.getElementById('first-name') as HTMLInputElement | null;
+        const lastNameInput = document.getElementById('last-name') as HTMLInputElement | null;
```

Comment:
Unless you did it on purpose, I'd suggest [controlling the inputs with react state](https://react.dev/reference/react-dom/components/input#controlling-an-input-with-a-state-variable) or [getting the values from the form](https://react.dev/reference/react-dom/components/form#handle-form-submission-on-the-client) rather than querying the HTML elements and relying on global IDs.